daemons/controld/controld_membership.c

```diff
@@ -403,7 +403,9 @@ crm_update_quorum(gboolean quorum, gboolean force_update)
         pcmk__panic(__func__);
     }
 
-    if (AM_I_DC && (force_update || fsa_has_quorum != quorum)) {
+    if (AM_I_DC
+        && ((pcmk_is_set(controld_globals.flags, controld_has_quorum) != quorum)
```

Comment:
The possibilities here are `true == true`, `false == false,`, `true == false`, and `false == true`, which seems safe. I'll update to add the two unequal cases explicitly though

Reply:
Yeah I find when comparing booleans there's always some corner case or future change that bites you. :) It would be more reassuring if bool were a true boolean type that couldn't hold anything but true or false.